helm-chart/binderhub/templates/secret.yaml

```
@@ -21,6 +21,15 @@ stringData:
   {{- /* Glob files to allow them to be mounted by the binderhub pod */}}
   {{- /* key=filename: value=content */}}
   {{- (.Files.Glob "files/*").AsConfig | nindent 2 }}
+
+  {{- with include "binderhub.extraFiles.stringData" .Values.extraFiles }}
+  {{- . | nindent 2 }}
+  {{- end }}
+
+{{- with include "binderhub.extraFiles.data" .Values.extraFiles }}
+data:
+  {{- . | nindent 2 }}
+{{- end }}
```

Did you know/consider this option? The helpers seems general enough to function without being hard-copied to this helm chart, and are practically available in the helm chart already.

```suggestion
  {{- with include "jupyterhub.extraFiles.stringData" .Values.extraFiles }}
  {{- . | nindent 2 }}
  {{- end }}

{{- with include "jupyterhub.extraFiles.data" .Values.extraFiles }}
data:
  {{- . | nindent 2 }}
{{- end }}
```

---

I will try that! I thought of it, but then wasn't sure about versions, etc. I will try it that way tomorrow

---

Since binderhub pins the version of JupyterHub being the dependency, this feels quite safe. Since the helpers are general enough, I'm not worried about having binderhub starting to use this - at least no more worried than needing to update the code in two places if it improves.

---

Can I just check, is this considered a supported use of Z2JH chat features (which therefore will be maintained in accordance with semver)?

---

I have two answers for that:

1. I would not consider any helpers in any chart public APIs, unless thoroughly documented to be otherwise
2. I would not consider binderhub an 'external' project, so I'm okay using these kinds of internal utilities across our _internal_ projects, especially with the strict dependency pin we use in binderhub. If jupyterhub ever breaks this, we can vendor the previous version with a copy/paste or update.

---

tested and works great with the imported helpers

---

I appreciate the answers above about "not documented -> no commitment" and the distinction about what we recommend using to others and what we do internally.

@manics do you see an action point that would make you feel comfortable around this? Perhaps a declaration of intent for how helpers are used in z2jh? I think being explicit about such intent is fine, especially since I'm unsure about what helpers we should couple to be tracked by the changelog etc.

---

It's related to the idea behind dogfooding: https://en.m.wikipedia.org/wiki/Eating_your_own_dog_food

We know JupyterHub and Z2JH are used and extended by lots of users, many who we have no idea about, and BinderHub is a prime demonstration of how to build on top of Z2JH. If we're running into issues with extending Z2JH then it's probable that others will too. I think we should therefore aim to do whatever we'd tell others to do. I can't tell from this discussion what the feeling was- say it's acceptable for everything else to do this with the proviso that it's not a standard API, or try and keep it hidden from everyone?

Given the time pressure it's fine to merge this, but if the conclusion of the above discussion is that this shouldn't be copied by other applications then maybe create an issue for how to come up with a public solution?

---

Just chiming it that while I don't consider them public _now_ that doesn't mean they necessarily shouldn't be - if we want to support downstream charts using extraFiles, I think the first step is adding them to the z2jh docs.